File: go/leap/leap.go
package leap

func IsLeapYear(i int) bool {
	switch {
	case i%4 == 0 && i%400 == 0:
		return true
	case i%4 == 0 && i%100 == 0:
		return false
	case i%4 == 0:
		return true
	default:
		return false
	}
}

File: ecmascript/leap/babel-output/leap.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = isLeapYear;

function isLeapYear(year) {
  // year divisible by 4, but not 100 unless also divisible by 400
  return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
}

module.exports = exports["default"];